src/test/scala/org/ergoplatform/nodeView/wallet/persistence/WalletRegistrySpec.scala

```diff
@@ -204,6 +204,27 @@ class WalletRegistrySpec
     }
   }
 
+  it should "update scans correctly" in {
+    val appId1: ScanId = ScanId @@ 21.toShort
+    val appId2: ScanId = ScanId @@ 22.toShort
+
+    forAll(trackedBoxGen) { tb0 =>
+      withHybridStore(10) { store =>
+        val tb1 = tb0.copy(scans = Set(appId1, appId2), spendingHeightOpt = None, spendingTxIdOpt = None)
+
+        val reg = new WalletRegistry(store)(ws)
+        WalletRegistry.putBox(emptyBag, tb1).transact(store)
```

Comment:
Looks like a backdoor access to store. Why not use registry to put the box?

Reply:
Registry is a versioned storage, putBox just puts ke-value pairs into a bag for a versioned update.